server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -683,12 +686,18 @@ private void announceExistence() {
 
         @Override
         public void lostLock(final LockLossReason reason) {
-          Halt.halt(serverStopRequested ? 0 : 1, () -> {
-            if (!serverStopRequested) {
-              log.error("Lost tablet server lock (reason = {}), exiting.", reason);
-            }
-            gcLogger.logGCInfo(getConfiguration());
-          });
+          if (isShutdownRequested()) {
+            LOG.warn(
```

Comment:
Tablet server should always halt when its lock is lost as the manager may reassign its tablets after the lock is lost.

Reply:
That makes sense to do, however the check is too broad.   There is a period of time where isShutdownRequested is true but the tablet server has not deleted its lock and is working on shutting down and could even be stuck.  If its looses its lock during this time period it should still halt.

Reply:
Addressed in 0d5f014